examples/platforms/emsk/uart.c

```diff
@@ -0,0 +1,169 @@
+/*
+ *  Copyright (c) 2017, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file implements the OpenThread platform abstraction for UART communication.
+ *
+ */
+
+#include <stddef.h>
+
+#include "openthread/types.h"
+#include <common/code_utils.hpp>
+#include "openthread/platform/uart.h"
+#include "platform-emsk.h"
+
+#include <stdio.h>
+#define DBG(fmt, ...)   printf(fmt, ##__VA_ARGS__)
+
+enum
+{
+    kUartId = BOARD_CONSOLE_UART_ID,
+    kBaudRate = BOARD_CONSOLE_UART_BAUD,
+    kReceiveBufferSize = 128,
+};
+
+static void processReceive(void);
+static void processTransmit(void);
+
+static const uint8_t *sTransmitBuffer = NULL;
+static uint16_t sTransmitLength = 0;
+
+static uint8_t sReceiveBuffer[kReceiveBufferSize];
+static uint16_t sReceiveHead = 0;
+
+static DEV_UART *consoleUart;
+
+ThreadError otPlatUartEnable(void)
+{
+    int32_t stateUart = 0;
+    ThreadError error = kThreadError_Drop;
+
+    /* UART in embARC */
+    consoleUart = uart_get_dev(BOARD_CONSOLE_UART_ID);
+
+    VerifyOrExit(!(consoleUart == NULL), DBG("Console UART is missing.\r\n"));
+
+    stateUart = consoleUart->uart_open(BOARD_CONSOLE_UART_BAUD);
+    if (stateUart == E_OPNED)
+    {
+        consoleUart->uart_control(UART_CMD_SET_BAUD, (void *)(BOARD_CONSOLE_UART_BAUD));
+        error = kThreadError_None;
+        DBG("Set Console UART Baudrate to 115200.\r\n");
```

Comment:
Yes, but you reference it as `BOARD_CONSOLE_UART_BAUD`.  Since it is not reference specifically in the function call, it probably shouldn't be in the print either, this way if the compiler macro is changed then the debug text will still be accurate.  I'd suggest something like this (assuming your debug functions support printf like syntax):
```
DBG("Set Console UART Baudrate to %d.\r\n", BOARD_CONSOLE_UART_BAUD);
```